File: flutter_app/lib/core/error/exceptions.dart
/// Excepciones técnicas lanzadas por la capa `data` (datasources).
/// El repositorio las traduce a `Failure` de dominio.
class ServerException implements Exception {
  final String message;
  ServerException([this.message = 'Error del servidor']);
}

class NetworkException implements Exception {
  final String message;
  NetworkException([this.message = 'Error de red']);
}

File: react_native_app/src/core/error/exceptions.ts
// Excepciones técnicas lanzadas por la capa `data` (datasources).
// El repositorio las traduce a `Failure` de dominio.

export class ServerException extends Error {
  constructor(message = 'Error del servidor') {
    super(message);
    this.name = 'ServerException';
  }
}

export class NetworkException extends Error {
  constructor(message = 'Error de red') {
    super(message);
    this.name = 'NetworkException';
  }
}
